source/common.ts

```diff
@@ -0,0 +1,51 @@
+import { createOwnCloudFileDetailProperty } from './helper';
+import { NextcloudClientInterface } from './types';
+import * as Path from 'path';
+
+export async function getFileOrFolderCreator(fullPath) : Promise<string> {
+  const self: NextcloudClientInterface = this;
+
+  let result = null;
+
+  const baseFolder = Path.dirname(fullPath);
+  const fileOrFolderName = Path.basename(fullPath);
+
+  if (await self.exists(baseFolder)) {
```

Comment:
Since you're using try/catch below, maybe that'd also be an option here?
```
try {
    assert(await self.exists(baseFolder));
    …
    result = await self.getObjectCreator();
} catch {
    return null;
}